addons/graph/src/presenter.js

```diff
@@ -1725,84 +1733,150 @@ function Addongraph_create(){
+        presenter.GSAcounter = ++itemIndex;
+        presenter.isGradualShowAnswersActive = true;
+    }
+    
+    presenter.gradualHideAnswers = function () {
+        if (presenter.configuration.isNotActivity || !presenter.isGradualShowAnswersActive) {
+            return;
+        }
+
+        presenter.$view.find(".graph_value_container").each(function (index, element) {
+            presenter.removeAnswerFromGraph(index, element);
+        });
+
+        presenter.isGradualShowAnswersActive = false;
+        presenter.currentData = [];
+        presenter.GSAcounter = 0;
+    }
+
+    presenter.setCurrentState = function () {
+        presenter.$view.find(".graph_value_container").each(function (index, element) {
+            presenter.currentData[index] = $(element).attr("current-value");
+        });
+    }
+
+    presenter.shouldStopAction = function () {
+        return presenter.isDisplayingAnswers() || presenter.errorMode;
+    }
+
+    presenter.isDisplayingAnswers = function () {
+        return presenter.isShowAnswersActive || presenter.isGradualShowAnswersActive;
+    }
+
+    presenter.addAnswerToGraph = function (index, element) {
+        const currentValue = presenter.configuration.answers[index],
+            valueContainer = $(element),
+            $columnContainer = valueContainer.parent('').find('.graph_column_container_interactive');
+        if(currentValue >= 0) {
+            valueContainer.css({
+                bottom: (presenter.drawingXPosition - Math.round(presenter.axisXLine.height() / 2)) + 'px',
+                height: parseFloat(currentValue / presenter.absoluteRange) * 100 + '%',
+                top: ''
+            });
+            $columnContainer.addClass('graph_column_container_show_answers');
+            valueContainer.addClass('graph_show_answers');
+        } else if (currentValue < 0) {
+            valueContainer.css({
+                height: parseFloat(currentValue * -1 / presenter.absoluteRange) * 100 + '%',
+                top: (presenter.chartInner.height() - presenter.drawingXPosition + Math.round(presenter.axisXLine.height() / 2)) + 'px',
+                bottom: ''
+            });
+            $columnContainer.addClass('graph_column_container_show_answers');
+            valueContainer.addClass('graph_show_answers');
+        }
+    }
+
+    presenter.removeAnswerFromGraph = function (index, element) {
+        const currentValue = presenter.currentData[index],
+            valueContainer = $(element),
+            $columnContainer = valueContainer.parent('').find('.graph_column_container_interactive');
+
+        if(currentValue >= 0) {
+            valueContainer.css({
+                bottom: (presenter.drawingXPosition - Math.round(presenter.axisXLine.height() / 2)) + 'px',
+                height: parseFloat(currentValue / presenter.absoluteRange) * 100 + '%',
+                top: ''
+            });
+            $columnContainer.removeClass('graph_column_container_show_answers');
```

Comment:
To usuwanie klas (i nadawanie w metodzie `addAnswerToGraph`) może by umiejscowić po if'ie, aby uniknąć powtórzeń?